### PR TITLE
Qa canonizer sania

### PR DESCRIPTION
**Pull request by saniatg**
https://canonizer.atlassian.net/browse/CAN-64
Forgot password automated test scripts

https://canonizer.atlassian.net/browse/CAN-500
Create new topic automated test scripts

### CODE REVIEW

File: CanonizerAcoountSettingPage.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
+from time import sleep
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: File name is incorrect. It should be Account where it is `Acoount`

File: CanonizerAcoountSettingPage.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+            Return the result to the main page.
+        """
+        self.hover(*AccountSettingPageIdentifier.CLICK_ON_DROPDOWN)
+        self.find_element(*AccountSettingPageIdentifier.CLICK_ON_DROPDOWN).click()
+        self.hover(*AccountSettingPageIdentifier.ACCOUNT_SETTING_BUTTON)
+        self.find_element(*AccountSettingPageIdentifier.ACCOUNT_SETTING_BUTTON).click()
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verifying_profile_button(self):
+        sleep(3)
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: Please remove sleep

File: CanonizerAcoountSettingPage.py [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+        self.hover(*AccountSettingPageIdentifier.PROFILE_BUTTON)
+        self.find_element(*AccountSettingPageIdentifier.PROFILE_BUTTON).click()
+        # sleep(3)
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verifying_social_oauth_verification(self):
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.SOCIAL_OAUTH_VERIFICATION)
+        self.find_element(*AccountSettingPageIdentifier.SOCIAL_OAUTH_VERIFICATION).click()
+        tittle = self.find_element(*AccountSettingPageIdentifier.SOCIAL_OAUTH_VERIFICATION).text
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: typo in the variable name.

File: CanonizerAcoountSettingPage.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+        print(tittle)
+        if tittle == 'Social Oauth Verification':
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: This should come from a common message file.

File: CanonizerAcoountSettingPage.py [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+            return CanonizerAccountSettingPage(self.driver)
+
+    def verifying_change_password(self):
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.CHANGE_PASSWORD)
+        self.find_element(*AccountSettingPageIdentifier.CHANGE_PASSWORD).click()
+        tittle = self.find_element(*AccountSettingPageIdentifier.CHANGE_PASSWORD).text
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: Typo in name the variable

File: CanonizerAcoountSettingPage.py [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+        print(tittle)
+        if tittle == 'Change Password':
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: `Change Password` should come from a common message file

File: CanonizerAcoountSettingPage.py [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+            return CanonizerAccountSettingPage(self.driver)
+
+    def verifying_nick_name(self):
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.NICK_NAME)
+        self.find_element(*AccountSettingPageIdentifier.NICK_NAME).click()
+        tittle = self.find_element(*AccountSettingPageIdentifier.NICK_NAME).text
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: typo in variable name

File: CanonizerAcoountSettingPage.py [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+        print(tittle)
+        if tittle == 'Nick Names':
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: Do we need to print? I think no

File: CanonizerAcoountSettingPage.py [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+        self.click_account_settings_page_button()
+        sleep(5)
+        self.find_element(*AccountSettingPageIdentifier.GENDER).click()
+        self.find_element(*AccountSettingPageIdentifier.UPDATE_BUTTON).click()
+
+        return CanonizerAccountSettingPage(self.driver)
+
+    #def Verify_the_functionality_of_selecting_the_DOB(self, dob):
+        self.click_account_settings_page_button()
+        sleep(3)
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: Please remove sleep from all the places

File: CanonizerAcoountSettingPage.py [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+
+    def test_Verify_entering_the_invalid_current_password(self, current):
+        self.click_account_settings_page_button()
+        sleep(5)
+        self.find_element(*AccountSettingPageIdentifier.CHANGE_PASSWORD).click()
+        self.find_element(*AccountSettingPageIdentifier.CURRENT_PASSWORD).send_keys(current)
+        self.find_element(*AccountSettingPageIdentifier.SAVE_BUTTON).click()
+        tittle = self.find_element(*AccountSettingPageIdentifier.CURRENT_PASSWORD_ERROR).text
+        print(tittle)
+        if tittle == 'Incorrect Current Password':
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py]: All the messages must come from a common file

**Review by akuks — CHANGES_REQUESTED** [review]

**Comment by saniatg** [comment]
@akuks All changes are done as mentioned by you.

File: CanonizerBrowsePage.py [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    def click_browse_page_button(self):
+        """
+        This function is to click on the Browse link
+        -> Hover to the Browse link
+        -> Find the element and click it
+
+        :return:
+            Return the result to the main page.
+        """
+    def Verify_Browse_page_is_opening_when_you_click_on_Browse(self):
```
Comment (akuks) [inline CanonizerBrowsePage.py]: function name should start with the smallcase

File: CanonizerBrowsePage.py [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+        self.hover(*BrowsePageIdentifiers.CLICK_ON_BROWSE)
+        self.find_element(*BrowsePageIdentifiers.CLICK_ON_BROWSE).click()
+        return CanonizerBrowsePage(self.driver)
+
+    def Verify_all_the_specified_fields_are_present_on_the_Browse_page(self):
```
Comment (akuks) [inline CanonizerBrowsePage.py]: function name should start with the smallcase

File: CanonizerBrowsePage.py [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+        self.Verify_Browse_page_is_opening_when_you_click_on_Browse()
+        sleep(2)
```
Comment (akuks) [inline CanonizerBrowsePage.py]: We are still using sleep?

File: CanonizerBrowsePage.py [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+        self.hover(*BrowsePageIdentifiers.SELECT_NAME_SPACE)
+        self.find_element(*BrowsePageIdentifiers.SELECT_NAME_SPACE).click()
+        self.find_element(*BrowsePageIdentifiers.ONLY_MY_TOPICS).click()
+        return CanonizerBrowsePage(self.driver)
+
+    def Check_for_the_validation_of_Select_Namespace_drop_down(self):
```
Comment (akuks) [inline CanonizerBrowsePage.py]: function name should start with the smallcase

File: CanonizerBrowsePage.py [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+        sleep(3)
```
Comment (akuks) [inline CanonizerBrowsePage.py]: Are we still using sleep?

**Review by akuks — CHANGES_REQUESTED** [review]

File: CanonizerBrowsePage.py (hunk 15 above)
Comment (saniatg) [reply CanonizerBrowsePage.py]: @akuks that was Sai's code. I have deleted that file now.

**Review by akuks — APPROVED** [review]

**Pull request merged**